custom_components/waste_collection_schedule/waste_collection_schedule/source/renoweb_dk.py

```diff
@@ -0,0 +1,125 @@
+""" Support for Renoweb waste collection schedule """
+
+import logging
+from typing import List
+import re
+from datetime import datetime
+import json
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+
+TITLE = "RenoWeb"
+DESCRIPTION = "RenoWeb collections"
+URL = "https://renoweb.dk"
+
+TEST_CASES = {
+    "test_01": {
+        "municipality": "frederiksberg",
+        "address": "Roskildevej 40",
+    },
+    "test_02": {
+        "municipality": "htk",
+        "address": "🤷",
+        "address_id": 45149,
+    },
+    "test_03": {
+        "municipality": "rudersdal",
+        "address": "Stationsvej 38",
+    }
+}
+
+_LOGGER = logging.getLogger('waste_collection_schedule.renoweb_dk')
+
+
+class Source:
+    """ Source class for RenoWeb """
+
+    _api_url: str
+    _address_id: int
+
+    def __init__(self, **kwargs):
```

Comment:
Please do not use **kwargs but each argument separately (use = None for optionals) The user should instantly get an error when passing wrong arguments

Reply:
Done, now also raises an exception if neither address or address_id is provided